### ISSUE TITLE
Alpha (Renyi) divergence ELBO

### ISSUE THREAD

**Issue by fehiepsi** [pyro-ppl/numpyro#386]
We think this is a  very nice feature to have and is a good practice to learn about how Elbo is implemented in NumPyro and how cool JAX is.

### References:
+ [Derivation](https://nbviewer.jupyter.org/gist/fehiepsi/7916400d8380cb05d21e859681b7a694) (skip non-reparameterizable part)
+ Pyro tutorial on ELBO: http://pyro.ai/examples/svi_part_iii.html#Setup (skip Non-reparameterizable Random Variables part).
+ [Pyro implementation of renyi_elbo](https://github.com/pyro-ppl/pyro/blob/dev/pyro/infer/renyi_elbo.py#L132) (ignore `score_function`, `entropy` because we only need to support reparameterizable models).
+ Rényi Divergence Variational Inference paper: https://arxiv.org/abs/1602.02311
+ The special case  `alpha=0` (i.e. order = 0) is equivalent to Importance Weighted lower bound: https://arxiv.org/abs/1509.00519

### What I need to know to implement this cool lower bound?
+ First, following the current elbo implementation in [elbo file](https://github.com/pyro-ppl/numpyro/blob/master/numpyro/infer/elbo.py), compute elbo of each particle (i.e. a sample from `guide`). This can be done by using JAX [vectorized map](https://jax.readthedocs.io/en/latest/jax.html#vectorization-vmap).
+ Following the above derivation (`f` there is elbo), compute weights of each particle and use JAX [stop_gradient](https://jax.readthedocs.io/en/latest/_autosummary/jax.lax.stop_gradient.html#jax.lax.stop_gradient) to prevent gradient flow through the weights.

### What I can take away when implementing this feature?
+ Know how elbo / important weighted / alpha divergence lower bounds are implemented.
+ Know how to vectorize your computation without manipulating the batch dimension.
+ Know how to skip a gradient flow in JAX.
+ And the most important thing: know how simple it is to implement an objective function in NumPyro which works for many models.

Please comment below any related question you have, we are happy to help you go through this issue. :)

**Comment by TuanNguyen27** [comment]
Also interested in taking this issue ! :D
